### PR TITLE
Added Definition and perspective for Branching

### PR DESCRIPTION
**Pull request by EslemOuederni**
## This PR fixes...

Adds a definition as well as my perspective of branching as a developer.
## What I did...

- Searched for git related words in the dictionary
- Changed the definition, perspective of branching, while also adding the source url

## I learned...

I learned to read the contribution guideline carefully.

### CODE REVIEW

File: src/assets/content/dictionary/branching.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 ---
 title: branching
-definition: ''
-perspectives: []
+definition: 'Branching in git, is to diverge from your main line of development and to continue working without affecting the main line.'
```
Comment (alliequintano) [inline src/assets/content/dictionary/branching.md:3]:
```suggestion
definition: 'Branching in git is to diverge from your main line of development and to continue working without affecting the main line.'
```

File: src/assets/content/dictionary/branching.md [hunk 2]
```diff
@@ -1,6 +1,10 @@
...
+sources:
+- sourceurl: https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell
+perspectives:
+- meaning: a way to help creating new features and fix bugs without affecting the main branch
```
Comment (alliequintano) [inline src/assets/content/dictionary/branching.md:7]:
```suggestion
- meaning: a way to help create new features and fix bugs without affecting the main branch
```

**Review by alliequintano — CHANGES_REQUESTED** [review]
Looks pretty good, please see my suggestions for some small improvements. Thank you!

**Review by EslemOuederni — COMMENTED** [review]
Thank you for the requested changes.

**Pull request closed without merging**